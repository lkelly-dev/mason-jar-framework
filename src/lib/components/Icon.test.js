import React from 'react';
import { shallow } from 'enzyme';
import { Icon } from "../index.js";

describe('Icon', () => {
  it('matches snapshot', () => {
    const component = shallow(<Icon></Icon>);
    expect(component).toMatchSnapshot();
  });

  it('has expected props', () => {
    const iconName = 'anything-at-all'; 
    const component = shallow(
      <Icon name={ iconName } className={ cn } what="ever" />
    );
    const classNameArray = component.prop('className').split(' ');
    expect(component.type()).toBe('span')
    expect(component.prop('what')).toBe('ever');
    expect(classNameArray).toContain('fe');
    expect(classNameArray).toContain(`fe-${iconName}`);
    expect(classNameArray).toContain(cn);
    expect(classNameArray).toHaveLength(3);
  });
  it('can pass custom element type', ()=>{
    const elType='i';
    const component = shallow(<Icon as={ elType }/>);
    expect(component.type()).toBe(elType)
  })

  it('appends data-attributes in debugMode', () => {
    const Context = React.createContext({debugMode: true});
    const component = shallow(
      <Icon
        context={ Context }
        name={ cn }
        className={ cn }
        what="ever">
      </Icon>
    );
    expect(component.prop('data-what')).toBe('ever');
    expect(component.prop('data-name')).toBe(cn);
    expect(component.prop('data-className')).toBe(cn);
  })
});
