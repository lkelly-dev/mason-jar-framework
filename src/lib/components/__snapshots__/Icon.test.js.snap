// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon matches snapshot 1`] = `
<span
  className="fe fe-undefined"
/>
`;
